temporal/fx.go

```diff
@@ -612,144 +606,126 @@ func ApplyClusterMetadataConfigProvider(
 	defer clusterMetadataManager.Close()
 
 	clusterData := config.ClusterMetadata
-	for clusterName, clusterInfo := range clusterData.ClusterInformation {
-		if clusterName != clusterData.CurrentClusterName {
-			logger.Warn(
-				"ClusterInformation in ClusterMetadata config is deprecated. "+
-					"Please use TCTL admin tool to configure remote cluster connections",
-				tag.Key("clusterInformation"),
-				tag.ClusterName(clusterName),
-				tag.IgnoredValue(clusterInfo))
-			continue
-		}
+	if clusterData.InitialFailoverVersion == 0 || len(clusterData.InternalRPCAddress) == 0 {
+		return config.ClusterMetadata, config.Persistence, fmt.Errorf("ClusterInformation config is deprecated. Please set InitialFailoverVersion and InternalRPCAddress in ClusterMetadata config")
+	}
```

Comment:
Can we read from db and only require the value from config if there is no db record for current cluster?
I'm trying to make this upgrade less disruptive.

Reply:
Good point. I will backfill this from db.